File: java/颠倒整数.java
class Solution {
    public int reverse(int x) {
        String s = String.valueOf(x);
        String reverse = new StringBuffer(s).reverse().toString();
        int index = reverse.indexOf("-");
        int i = 0;
        if(index != -1 ){
            String newReverse = reverse.substring(0,reverse.length()-1);
            try {
                i = Integer.parseInt(newReverse);
                i = -i;
            }catch(NumberFormatException e){
                i = 0;
            }
        }
        else{
            try{
                i = Integer.parseInt(reverse);
            }catch(NumberFormatException e){
                i = 0;
            }
        }
        return i;
    }
}

File: 初级算法/字符串/颠倒整数.cpp
class Solution {
public:
    int reverse(int x) {
        int flag = 0;
        if(x < 0){
            flag = 1;
            x = 0 - x;
        }
        string result = to_string(x);
        std::reverse(result.begin(),result.end());
        try{
            x = stoi(result);
        }catch(exception &e){ //防止int x超出边界
            return 0;
        }
        if(flag == 1)
            x = 0 - x;
        return x;
        
    }
};